apps/user-profile/src/app/pages/settings-page/settings-mailing-lists/settings-mailing-lists.component.ts

```diff
@@ -27,25 +31,100 @@ export class SettingsMailingListsComponent implements OnInit {
   index: number;
   filteredVos: Vo[] = [];
   loading: boolean;
+  selectedVo: string = null;
+  selectedResource: string = null;
+  changeOptOut: string;
 
   constructor(
     private store: StoreService,
     private usersManagerService: UsersManagerService,
     private membersService: MembersManagerService,
+    private vosManagerService: VosManagerService,
     private resourcesManagerService: ResourcesManagerService,
-    private attributesManagerService: AttributesManagerService
+    private attributesManagerService: AttributesManagerService,
+    private route: ActivatedRoute,
+    private router: Router,
+    private location: Location,
+    private notificator: NotificatorService,
+    private translate: TranslateService
   ) {}
 
+  ngOnDestroy(): void {
+    // to avoid keeping query params in the url when going to a different part of the website,
+    //  a better solution probably exists.
+    void this.router.navigate([], {
+      relativeTo: this.route,
+      queryParams: { vo: null, resource: null },
+      replaceUrl: true,
+    });
+  }
+
   ngOnInit(): void {
+    let voShortName;
+    let resourceName;
+    let action;
```

Comment:
are this variables necassary? why dont you just assign the query params to the component variables?